package-res/resources/web/pentaho/type/property.js

```diff
@@ -0,0 +1,1331 @@
+           * an error is thrown.
+           *
+           * The default (root) `countMax` attribute value is `Infinity`.
+           *
+           * @name countMax
+           * @memberOf pentaho.type.Property.Type#
+           * @type undefined | number | pentaho.type.PropertyDynamicAttribute.<number>
+           *
+           * @see pentaho.type.Complex#countRange
+           * @see pentaho.type.spec.IPropertyTypeProto#countMax
+           */
+          countMax: {
+            value: Infinity,
+            cast: castCount,
+            combine: function(baseEval, localEval) {
+              return function() {
+                return Math.min(baseEval.call(this), localEval.call(this));
+              };
+            }
+          },
+
+          /**
+           * Evaluates the value of the `isApplicable` attribute of a property of this type
+           * on a given owner complex value.
+           *
+           * @name applicableEval
+           * @memberOf pentaho.type.Property.Type#
+           * @param {pentaho.type.Complex} owner The complex value that owns a property of this type.
+           * @return {boolean} The evaluated value of the `isApplicable` attribute.
+           *
+           * @ignore
+           */
+
+          /**
+           * Gets or sets a value or function that indicates if properties of this type are applicable.
+           *
+           * When a property is not (currently) applicable, then it does not apply,
+           * it does not make sense in a certain situation.
+           * It may only be applicable when another property of the complex type has a specific value, for example.
+           *
+           * When a property is not currently applicable, its value is not significant,
+           * and, as such, any validations concerning it are not performed.
+           *
+           * ### This attribute is *Dynamic*
+           *
+           * When a _dynamic_ attribute is set to a function,
+           * it can evaluate to a different value for each complex instance.
+           *
+           * When a _dynamic_ attribute is set to a value other than a function or a {@link Nully} value,
+           * its value is the same for every complex instance.
+           *
+           * ### This attribute is *Monotonic*
+           *
+           * The value of a _monotonic_ attribute can change, but only in some, predetermined _monotonic_ direction.
+           *
+           * In this case, a _property type_ marked as _applicable_ can later be marked as _not applicable_.
+           * However, a _property type_ marked as _not applicable_ can no longer go back to being _applicable_.
+           *
+           * Because this attribute is also _dynamic_,
+           * the actual `isApplicable` values are only known
+           * when evaluated for specific complex instances.
+           * As such, ensuring monotonic changes is deferred until evaluation.
```

Comment:
Suggest:

This behavior ensures that monotonic changes are deferred until evaluation.